Xamarin.Essentials/WebAuthenticator/WebAuthenticator.ios.tvos..cs

```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using AuthenticationServices;
+using Foundation;
+#if __IOS__
+using SafariServices;
+#endif
+using UIKit;
+
+namespace Xamarin.Essentials
+{
+    public static partial class WebAuthenticator
+    {
+        static TaskCompletionSource<AuthResult> tcsResponse;
+        static UIViewController currentViewController;
+        static Uri redirectUri;
+
+        internal static async Task<AuthResult> PlatformAuthenticateAsync(Uri url, Uri callbackUrl)
+        {
+            tcsResponse = new TaskCompletionSource<AuthResult>();
```

Comment:
We probably should cancel any previous tasks - just in case. Since that is static.

Reply:
Added code to first cancel any previous tasks anytime we start a new auth flow on all platforms.